File: playground/read.c
#include <unistd.h>
#include <fcntl.h>
#include "myHeader.h"

int main() {
    char* fileName = "hello.txt";
    printf("%s\n",fileName);
    printf("PI = %f\n",PI);
    int fd = open(fileName, O_RDWR);
    if(fd == -1) {
        printf("\nError opening file!!\n");
        return 1;
    } else {
        printf("\nFile %s opened sucessfully!\n", fileName);
    }
    char buffer[1024];

    int bytesRead = read(fd, buffer, sizeof(buffer));


    printf("%d bytes read!\n", bytesRead);
    printf("File contents: %s\n", buffer);

    return 0;
}

File: playground/read.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

public class read {
    public static void main(String[] args) {
        try {
            String fileName = "hello.txt";
                   FileReader fileReader = new FileReader(fileName);
                BufferedReader bufferedReader = new BufferedReader(fileReader);
    
                String line;
                // Read the file line by line
                while ((line = bufferedReader.readLine()) != null) {
                    System.out.println(line);
                }
    
                // Close the file
                bufferedReader.close();

        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}